docs/couchdb-quirks.md

```diff
@@ -0,0 +1,66 @@
+# CouchDB Quirks
+
+## Mango indexes
+
+### Exists operator
+
+The [`$exists`
+operator](http://docs.couchdb.org/en/2.1.1/api/database/find.html#condition-operators)
+can be used with a mango index for the `true` value, but not for the `false`
+value. For `false`, a more heavy solution is required: [a partial
+index](http://docs.couchdb.org/en/2.1.1/api/database/find.html#find-partial-indexes).
+
+### Index selection
+
+CouchDB may accept or refuse to use a mango index for a query, with obsure
+reasons. In general, you can follow these two rules of thumb:
+
+1. An index on the fields `foo, bar, baz` can be used only to fetch documents
+   where `foo`, `bar`, and `baz` exist. It means that a query that filters only
+   on the value on `foo` won't use the mango index, because it can miss a
+   document where `foo` has the expected value but without `bar` or `baz`. If
+   you know that all the documents that you want have the `bar` and `baz`
+   fields, you can just add two filters `$exists: true` (one for `bar`, the
+   other for `baz`).
+
+2. You should use exactly the same sequence of fields for creating the index
+   and the `sort` operator of the query. If you have an index on `os, browser,
+   ip` for the `io.cozy.sessions.logins`, and you want to have all the
+   documents for a login from `windows`, sorted by `browser`, you can use the
+   index, but you should use `os, browser, ip` for the sort (or at least `os,
+   browser`, even if it is seems to weird to sort on `os` when all the sorted
+   documents will have the same value, `windows`).
```

Comment:
May be add a sentence about how `sort`  is essentially useless when you use_index except for setting it do descending ?